futures-core/src/poll.rs

```diff
@@ -38,8 +38,48 @@ impl<T> Poll<T> {
     }
 }
 
+impl<T, E> Poll<Result<T, E>> {
+    /// Convenience for with with a `PollResult` as a `Result`
+    pub fn ok(self) -> Result<Poll<T>, E> {
+        match self {
+            Poll::Pending => Ok(Poll::Pending),
+            Poll::Ready(Ok(t)) => Ok(Poll::Ready(t)),
+            Poll::Ready(Err(e)) => Err(e),
+        }
+    }
+}
+
 impl<T> From<T> for Poll<T> {
     fn from(t: T) -> Poll<T> {
         Poll::Ready(t)
     }
 }
+
+/// Shorthand for a `Poll<Result<_, _>>` value.
+pub type PollResult<T, E> = Poll<Result<T, E>>;
+
```

Comment:
It seems like we should also have a macro here for propagating `Pending` that works on non-`Result`y `Poll`s.